var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -13,53 +13,106 @@ class Mercury(CMakePackage):
     git = 'https://github.com/mercury-hpc/mercury.git'
 
     version('develop', branch='master', submodules=True)
+    version('1.0.1', tag='v1.0.1', submodules=True)
     version('1.0.0', tag='v1.0.0', submodules=True)
     version('0.9.0', tag='v0.9.0', submodules=True)
 
-    variant('cci', default=False, description='Use CCI for network transport')
     variant('bmi', default=False, description='Use BMI for network transport')
-    variant('fabric', default=True, description='Use libfabric for network transport')
+    variant('cci', default=False, description='Use CCI for network transport')
+    variant('mpi', default=False, description='Use MPI for network transport')
+    variant('ofi', default=True,  description='Use libfabric plugin')
+    variant('sm',  default=False, description='Use shared-memory plugin')
+    variant('opa', default=False, description='Use OpenPA for atomics')
+    variant('boost', default=True, description='Use BOOST preprocessor macros')
     variant('selfforward', default=True,
             description='Mercury will short-circuit operations' +
                         ' by forwarding to itself when possible')
-# NOTE: the 'udreg' variant requires that the MPICH_GNI_NDREG_ENTRIES=1024
-#   environment variable be set at run time to avoid conflicts with
-#   Cray-MPICH if libfabric and MPI are used at the same time
+    # NOTE: the 'udreg' variant requires that the MPICH_GNI_NDREG_ENTRIES=1024
+    #   environment variable be set at run time to avoid conflicts with
+    #   Cray-MPICH if libfabric and MPI are used at the same time
     variant('udreg', default=False,
             description='Enable udreg on supported Cray platforms')
 
-    depends_on('cci@master', when='+cci', type=('build', 'link', 'run'))
-    depends_on('libfabric', when='+fabric', type=('build', 'link', 'run'))
-    depends_on('bmi', when='+bmi', type=('build', 'link', 'run'))
-    depends_on('openpa', type=('build', 'link', 'run'))
+    depends_on('cmake@2.8.12.2:', type='build')
+    # depends_on('bmi', when='+bmi')  # TODO: add BMI package
+    # depends_on('cci', when='+cci')  # TODO: add CCI package
```

Comment:
These packages have never existed in Spack, so I commented out the dependencies but left the variants in place. This way, if someone has these libraries installed on their system, they can still build `mercury+bmi+cci`. Someone should add packages for these dependencies someday, but not me, and not today.

Reply:
Normally I'd say it isn't bad to have the `depends_on`: if the user enables `+cci` and it isn't available then there would be an error (which could be considered good in the sense that it alerts the user to the lack of built-in Space support).

However, this would allow the user to make use of system-installed `cci`.